loops_lab.js

```diff
@@ -0,0 +1,222 @@
+let max = 10;
+let counter = 0;
+while (counter <= max) {
+  if (counter % 2 === 0) { console.log(counter + ' is even');
+  }   else {
+    console.log(counter + ' is odd');
+  }
+  counter++;
+}
+//
+console.log('----- #2 for loop -----');
+let max1 = 10;
+for (let counter1 = 0; counter1 <= max1; counter1++) {
+  if (counter1 % 2 === 0) { console.log(counter1 + ' even');
+  }  else {
+    console.log(counter1 + ' odd');
+  }
+}
+//
+// // 3. Write a `while` loop and a `for` loop that takes the variable "x" and iterates over all numbers from 0 to "x".
+// // For each iteration of the loop, it will multiply the number by 9 and log the result (e.g. "2 * 9 = 18").
+console.log('----- #3 while loop "cannot get the numbers to ascend" -----');
+// // cannot get the numbers to ascend
+let x = 11;
+let xx = 0;
+while (xx <= x){ //&& 0 <= x)
+
+  let num2 = xx * 9;
+  console.log(xx + ' * 9 = ' + num2);
+  xx++;
+}
+//
+console.log('----- #3 for loop "cannot get the numbers to ascend" -----');
+// // cannot get the numbers to ascend
+// let xx1 = 0;
+for (let x1 = 0; x1 <= x; x1++) {
+  console.log( x1 + ' * 9 = ' + (x1*9));
+}
+
+
+// _Bonus_ think of another way to solve it.
+//         Hint
+//       Find the final number and increment the loop by 9.
+// let x = 10;
+// let num1 = [];
+// while (x >= num1*9) {
+//   console.log(x + ' * 9 = ' + num1);
+//   x--;
+// }
+
+// 4. Write code that prints/logs all the numbers that end in a 5 from 1 to 100, **exclusive**.
+console.log('----- #4 -----');
+let fives = 5;
+while (fives <= 100) {
+console.log(fives);
+fives += 10;
+}
+// 5. Without running/executing your code, how many times will the loop below run? Explain why.
+//
+// ```js
+// let i = 5;
+//
+// while (i > 3) {
+//     i += 1;
+// }
+//
+// // Your explanation here
+// ```
+console.log('----- #5 -----');
+console.log('This code will run as an infinte loop because "i" will always be greater than 3. ')
+
+// 6. Write a loop that uses console.log to log all the numbers from 1 to 100, with two exceptions. For numbers divisible by 3, log "Fizz" instead of the number, and for numbers divisible by 5 (and not 3), log "Buzz" instead.
+console.log('----- #6 -----');
+const num3 = 100;
+for (let counter = 1; counter <= num3; counter++) {
+  if ((counter % 3) === 0) {
+    console.log('Fizz');
+  } else if ((counter % 5) === 0) {
+    console.log('Buzz');
+  } else {
+    console.log(counter);
+  }
+}
+
+// 7. Modify your program to log "FizzBuzz", for numbers that are divisible by both 3 and 5 (still log "Fizz" or "Buzz" for numbers divisible by only one of those).
+console.log('----- #7 -----');
+const num4 = 100;
+for (let counter = 1; counter <= num4; counter++) {
+  if ((counter % 3) === 0 && (counter % 5) === 0) {
+    console.log('Fizz Buzz');
+  } else if ((counter % 3) === 0) {
+    console.log('Fizz');
+  } else if ((counter % 5) === 0) {
+    console.log('Buzz');
+  } else {
+    console.log(counter);
+  }
+}
+
+//
+// ### Bonus:
```

Comment:
Also try bonus questions next time. They are good practice to master what you learned in a lesson.

Reply:
I've worked on bonus questions in my spare time after submitting the lab (want to submit on time), I just haven't pushed them to GitHub for the last two labs we've had.